index.js

```diff
@@ -8,62 +8,36 @@ var platforms = {
 };
 
 
-exports.verifyPayment = function (platform, payment, cb) {
-	function syncError(error) {
-		process.nextTick(function () {
-			cb(error);
-		});
-	}
-
-	if (!payment) {
-		return syncError(new Error('No payment given'));
-	}
-
-	var engine = platforms[platform];
-
-	if (!engine) {
-		return syncError(new Error('Platform ' + platform + ' not recognized'));
-	}
-
-	engine.verifyPayment(payment, function (error, result) {
-		if (error) {
-			return cb(error);
+function makeMethod(methodName) {
+	exports[methodName] = function (platform, payment, cb) {
+		function asyncError(message) {
+			var error = new Error(message);
+			// this will cause the stack trace to start at the place where asyncError
+			// is called
+			error.captureStackTrace(error, asyncError);
+			process.nextTick(cb, error);
 		}
 
-		result.platform = platform;
```

Comment:
We lost this line now.

Reply:
ups, good catch, it looks like cancelSubsription didn't have that line in the first place and I haven't noticed since I don't really use that property in my project

might be a smart thing to add some unit tests ;)